### PR TITLE
Add bool vector coercion

### LINKED ISSUE #1355: vector of int/float -> vector of bool casts
From https://github.com/gpuweb/gpuweb/issues/753#issuecomment-763037442

Let's add `vecN<bool>(vecN<*>)` to the standard library!

### PR DESCRIPTION
**Pull request by dneto0**
Fixes #1355

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -2277,6 +2277,27 @@ Details of conversion to and from floating point are explained in [[#floating-po
   <thead>
     <tr><th>Precondition<th>Conclusion<th>Notes
   </thead>
+  <tr algorithm="vector coercion of unsigned integer to boolean">
+     <td>|e| : vec|N|&lt;u32&gt;
+     <td>`vec`|N|&lt;`bool`&gt;`(`|e|`)` : vec|N|&lt;bool&gt
+     <td>Component-wise coercion of a unsigned integer vector to a boolean vector.<br>
+         Component |i| of the result is `bool(`|e|`[`|i|`])`<br>
+         (OpINotEqual to compare |e| against a zero vector.)
```
Comment (dj2) [inline wgsl/index.bs:2285]: I don't think this line is right given we're doing component wise checks?

File: wgsl/index.bs [hunk 2]
```diff
@@ -2277,6 +2277,27 @@ Details of conversion to and from floating point are explained in [[#floating-po
...
+
+  <tr algorithm="vector coercion of signed integer to boolean">
+     <td>|e| : vec|N|&lt;i32&gt;
+     <td>`vec`|N|&lt;`bool`&gt;`(`|e|`)` : vec|N|&lt;bool&gt
+     <td>Component-wise coercion of a signed integer vector to a boolean vector.<br>
+         Component |i| of the result is `bool(`|e|`[`|i|`])`<br>
+         (OpINotEqual to compare |e| against a zero vector.)
```
Comment (dj2) [inline wgsl/index.bs:2292]: ditto

File: wgsl/index.bs [hunk 3]
```diff
@@ -2277,6 +2277,27 @@ Details of conversion to and from floating point are explained in [[#floating-po
...
+
+  <tr algorithm="vector coercion of floating point to boolean">
+     <td>|e| : vec|N|&lt;f32&gt;
+     <td>`vec`|N|&lt;`bool`&gt;`(`|e|`)` : vec|N|&lt;bool&gt
+     <td>Component-wise coercion of a floating point vector to a boolean vector.<br>
+         Component |i| of the result is `bool(`|e|`[`|i|`])`<br>
+         (OpFUnordNotEqual to compare |e| against a zero vector.)
```
Comment (dj2) [inline wgsl/index.bs:2299]: ditto

File: wgsl/index.bs (hunk 1 above)
Comment (dneto0) [reply wgsl/index.bs:2285]: OpINotEqual operates on vector operands, producing a vector result of the same component count, and "Results are computed per component."

File: wgsl/index.bs (hunk 3 above)
Comment (dneto0) [reply wgsl/index.bs:2299]: Same

File: wgsl/index.bs (hunk 2 above)
Comment (dneto0) [reply wgsl/index.bs:2292]: Same

**Review by dj2 — APPROVED** [review]

**Review by kvark — APPROVED** [review]

**Review by jdashg — APPROVED** [review]

**Comment by kvark** [comment]
Doesn't look controversial, merging

**Pull request merged**